lib/index_documents.rb

```diff
@@ -0,0 +1,65 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    links = message.payload["links"].try(:dup)
+
+    if links
+      links.each do |(k, vs)|
+        paths = vs.map { |v| base_path(v) }.compact
+        links[k] = paths
+      end
+
+      base_path = document_base_path(message)
+      update_index(base_path, links)
+    end
+
+    message.ack
+  rescue StandardError => e
+    Airbrake.notify_or_ignore(
+      error: e,
+      payload: message.payload,
+      headers: message.headers,
+      delivery_info: message.delivery_info
+    )
+    message.retry
+  end
+
+  private
```

Comment:
Our style guide says to outdent private (ie, don't have leading space for it here).  Also that it shouldn't have a blank line following it.